pkg/kubelet/kubelet_pods.go

```diff
@@ -1107,11 +1108,18 @@ func (kl *Kubelet) HandlePodCleanups() error {
 		klog.ErrorS(err, "Error listing containers")
 		return err
 	}
+
+	var wg sync.WaitGroup
 	for _, pod := range runningPods {
 		if _, found := desiredPods[pod.ID]; !found {
-			kl.podKiller.KillPod(&kubecontainer.PodPair{APIPod: nil, RunningPod: pod})
+			wg.Add(1)
+			go func(pod *container.Pod) {
+				defer wg.Done()
+				<-kl.podKiller.KillPod(&kubecontainer.PodPair{APIPod: nil, RunningPod: pod})
+			}(pod)
 		}
 	}
+	wg.Wait()
```

Comment:
do we need to put a timeout for this wait?

Reply:
We can't safely put a timeout, because the rest of the function depends on the pods being killed.

An e2e test is tricky, since we only see this issue under specific CI load.

Reply:
How about if we return error if we reach the timeout?

Reply:
Can KillPods ever time out completely? It shouldn't completely time out.